src/openrct2/interface/Screenshot.cpp

```diff
@@ -607,16 +524,140 @@ int32_t cmdline_for_screenshot(const char** argv, int32_t argc, ScreenshotOption
+        return -1;
+    }
+
+    int32_t exitCode = 1;
+    rct_drawpixelinfo dpi;
+    try
+    {
+        core_init();
+        bool customLocation = false;
+        bool centreMapX = false;
+        bool centreMapY = false;
+
+        const char* inputPath = argv[0];
+        const char* outputPath = argv[1];
+
+        gOpenRCT2Headless = true;
+        auto context = CreateContext();
+        if (!context->Initialise())
+        {
+            throw std::runtime_error("Failed to initialize context.");
+        }
+
+        drawing_engine_init();
+
+        context->LoadParkFromFile(inputPath);
+
+        gIntroState = INTRO_STATE_NONE;
+        gScreenFlags = SCREEN_FLAGS_PLAYING;
 
-    rct_palette renderedPalette;
-    screenshot_get_rendered_palette(&renderedPalette);
+        rct_viewport viewport{};
+        if (giantScreenshot)
+        {
+            auto zoom = std::atoi(argv[3]);
+            auto rotation = std::atoi(argv[4]) & 3;
+            viewport = GetGiantViewport(gMapSize, rotation, zoom);
+        }
+        else
+        {
+            int32_t resolutionWidth = std::atoi(argv[2]);
+            int32_t resolutionHeight = std::atoi(argv[3]);
+            int32_t customX = 0;
+            int32_t customY = 0;
+            int32_t customZoom = 0;
+            int32_t customRotation = 0;
+            if (argc == 8)
+            {
+                customLocation = true;
+                if (argv[4][0] == 'c')
+                    centreMapX = true;
+                else
+                    customX = std::atoi(argv[4]);
+
+                if (argv[5][0] == 'c')
+                    centreMapY = true;
+                else
+                    customY = std::atoi(argv[5]);
+
+                customZoom = std::atoi(argv[6]);
+                customRotation = std::atoi(argv[7]) & 3;
+            }
+
+            int32_t mapSize = gMapSize;
+            if (resolutionWidth == 0 || resolutionHeight == 0)
+            {
+                resolutionWidth = (mapSize * 32 * 2) >> customZoom;
+                resolutionHeight = (mapSize * 32 * 1) >> customZoom;
+
+                resolutionWidth += 8;
+                resolutionHeight += 128;
+            }
+
+            viewport.width = resolutionWidth;
+            viewport.height = resolutionHeight;
+            viewport.view_width = viewport.width;
+            viewport.view_height = viewport.height;
+            if (customLocation)
+            {
+                if (centreMapX)
+                    customX = (mapSize / 2) * 32 + 16;
+                if (centreMapY)
+                    customY = (mapSize / 2) * 32 + 16;
+
+                int32_t z = tile_element_height(customX, customY);
+                CoordsXYZ coords3d = { customX, customY, z };
 
-    WriteDpiToFile(outputPath, &dpi, renderedPalette);
+                CoordsXY coords2d = translate_3d_to_2d_with_z(customRotation, coords3d);
```

Comment:
If you change this to auto then my change that makes this screen coords will merge nicer.